support using `include` to insert a mixin for Jinja

matin opened the pull request:
Pug supports the ability to include a file that has mixins:
https://github.com/pugjs/pug/blob/master/packages/pug/test/cases/includes.pug

This PR adds that functionality for Jinja templates.

tmzhuang commented:
@matin why'd you close this request? I'd like the see the changed you implemented merged.

matin commented:
@tmzhuang I ended up continuing development on [my own fork](https://github.com/matin/pypugjs).

I cleaned up some of the code and added the ability to include markdown from separate files.

I only worked on the code related to Jinja2 and didn't have much motivation to work on the other templating engines because I only use Flask and am not as familiar with the other template engines / frameworks. It didn't make sense to issue a PR because I broke the code for the other template engines in the process.

matin commented:
On a side note, I actually think the right approach would be to add a plugin for the node (canonical) version of pugjs to precompile to Jinja2. You'd no longer have to worry about keeping up to date with the canonical spec and would gain a slight performance advantage (if you're into that sort of thing) by avoiding two rounds of template compilation during the request cycle. Precompiling the templates with node would also be more consistent with the process used for CSS frameworks like less, postcss, etc. and javascript precompilers like babel.
